packages/firestore-types/index.d.ts

```diff
@@ -492,6 +492,16 @@ export interface SetOptions {
    * untouched.
    */
   readonly merge?: boolean;
+
+  /**
+   * Changes the behavior of set() calls to only replace the specified field
```

Comment:
What if I pass `[]`?

Reply:
This is allowed since `set({}, merge:true)` is also a valid use. We originally enforced that at least one field was updated, but got user reports asking for us to drop this validation.